File: 169-majority-element/majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int major = nums[0], cnt = 0;

        for (int num : nums) {
            if (num == major) cnt++;
            else cnt--;

            if (cnt == 0) {
                major = num;
                cnt = 1;
            }
        }

        return major;
    }
}

File: 169-majority-element/majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
       int i = 0;
       int max = nums[i]; //6
       int cnt = 0;
       while(i< nums.size()){
        if(nums[i] == max){
            cnt++; //1
        }
        else{
            cnt--;
        }
        i++; //5
         if(cnt == 0){
            max = nums[i];
        }
       }
      

        return max;
    }
};